dask/array/routines.py

```diff
@@ -1065,6 +1065,23 @@ def nonzero(a):
         return (ind,)
 
 
+def _int_piecewise(x, *condlist, **kwargs):
+    return np.piecewise(
+        x, list(condlist), kwargs["funclist"],
+        *kwargs["func_args"], **kwargs["func_kw"]
+    )
+
+
+@wraps(np.piecewise)
+def piecewise(x, condlist, funclist, *args, **kw):
+    return map_blocks(
+        _int_piecewise,
+        x, *condlist,
+        dtype=x.dtype,
+        funclist=funclist, func_args=args, func_kw=kw
+    )
```

Comment:
For cosmetic reasons it might be nice to ensure that the keynames generated by this look good in the dashboard.  Currently I suspect that they are `_int_piecewise`.  It looks like this can be done with `token='piecewise'`

Reply:
Happy to add it.

That said, this doesn't work as well as we would like. ( https://github.com/dask/dask/issues/2705 )